### DISCUSSION TITLE
How Can I Implement Icon Click Events in the AppBar to Show Popups or Drawers?

### DISCUSSION THREAD

**Discussion by Thanawit007z** [mui/toolpad#4528] [Help and support]
![image](https://github.com/user-attachments/assets/a99c32ed-141e-40f3-9ce0-9addb1fdc4d5)

How can I add icons to the AppBar (highlighted in the image) and create an event such that when clicking on the icon, it shows either a popup or a drawer? Is this possible to implement right now?

**Comment by bharatkashyap** [comment]
You should be able to add a custom component with buttons to the `toolbarActions` slot and add this functionality on click of those buttons

**Reply by THANAWIT888** [reply]
Can you please give me an example

**Reply by bharatkashyap** [reply]
Here's an example: https://codesandbox.io/p/sandbox/beautiful-thunder-p37259 

I've added a `SettingsIcon` before the search field, which triggers a drawer menu on click (you can add any arbitrary content in its place)
